neuland/geobase/create_neuland_geo.C

```diff
@@ -0,0 +1,75 @@
+/******************************************************************************
+ *   Copyright (C) 2023 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2023 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "./helper_neuland_geometry.C"
+#include "TGeoManager.h"
+#include "TMath.h"
+#include <iomanip>
+#include <iostream>
+
+void create_neuland_geo(const Int_t nPlanes = 26, const Double_t distance = 0.0, const TString geoTag = "v2022.5")
+{
+    new FairGeoLoader("TGeo", "FairGeoLoader");
+    gGeoManager->SetName("NEULANDgeom");
+
+    TGeoVolume* volPaddle = BuildPaddleVolume();
+    TGeoRotation* rot0 = nullptr;
+    TGeoRotation* rot90 = new TGeoRotation();
+    rot90->RotateZ(90.);
+    TGeoRotation* rot45 = new TGeoRotation();
+    rot45->RotateY(45.);
+
+    TGeoVolume* volNeuland = new TGeoVolumeAssembly("volNeuland");
+    Int_t nindex = 0, nPlane = 0;
+    for (Double_t b = -nPlanes * gPaddleDistance + gPaddleDistance; b < nPlanes * gPaddleDistance;
+         b += gPaddleDistance * 2)
+    {
+        nPlane++;
+        for (Double_t a = -gPaddlesPerPlane * gPaddleDistance + gPaddleDistance; a < gPaddleDistance * gPaddlesPerPlane;
+             a += gPaddleDistance * 2)
+        {
+            nindex++;
+            if (nPlane % 2 == 1)
+            {
+                volNeuland->AddNode(volPaddle, nindex, new TGeoTranslation(0, a, b));
+            }
+            else
+            {
+                volNeuland->AddNode(volPaddle, nindex, new TGeoCombiTrans(a, 0, b, rot90));
+            }
+        }
+    }
+
+    TGeoVolume* top = new TGeoVolumeAssembly("TOP");
+    gGeoManager->SetTopVolume(top);
+    top->AddNode(volNeuland, 1, new TGeoCombiTrans(0., 0., distance + nPlanes + gPaddleDistance, rot0));
```

Comment:
Wait, what's the meaning of this:
```cpp
new TGeoCombiTrans(0., 0., distance + nPlanes + gPaddleDistance, rot0)
```
Especially the summation `distance + nPlanes + gPaddleDistance`? Currently to generate Neuland geo file, we use `/macros/r3b/geo/create_neuland_v3_geo.C`. Is this MACRO file based on that?

Reply:
I don't know about that, but you have the figures above to see the reference point (0,0,0)

Reply:
Adding the number of planes with a distance value looks really strange to me. It makes sense if it's `nPlaines*gPaddleDistance`.

Reply:
It looks this '+' is a typo which was not corrected in all geometry versions. It should be nPlanes * gPaddleDistance. I think this went unnoticed because in a simulation macro the distance is usually overwritten anyhow.

Reply:
I have removed it since the user will decide the position.